applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -723,34 +720,24 @@ CrBeamElement3D2N::CalculateInitialLocalCS() const
     KRATOS_CATCH("")
 }
 
-void CrBeamElement3D2N::CalculateTransformationMatrix(
-    BoundedMatrix<double, CrBeamElement3D2N::msElementSize,
-    CrBeamElement3D2N::msElementSize>& rRotationMatrix,
-    Vector& Bisectrix, Vector& VectorDifference)
+Vector CrBeamElement3D2N::GetIncrementDeformation() const
 {
-
-    KRATOS_TRY
-    // update local CS
-    Matrix aux_rotation_matrix =
-        UpdateRotationMatrixLocal(Bisectrix, VectorDifference);
-
-    rRotationMatrix = ZeroMatrix(msElementSize);
-    // Building the rotation matrix for the local element matrix
-    AssembleSmallInBigMatrix(aux_rotation_matrix, rRotationMatrix);
+    KRATOS_TRY;
+    return mDeformationCurrent - mDeformationPrevious;

```

Comment:
why do you store the mDeformationCurrent? you can always recompute it...

Reply:
after speaking with @KlausBSautter 
We can **NOT** recompute this (believe me, we spend a LOT of time trying!)! This is the delta-deformation between the iterations!
It cannot be recomputed because in the NR-Strategy `UpdateDataBase` is called before `FinalizeNonLinIteration`
Hence computing it on the fly would always get the delta between the steps but not the iterations
https://github.com/KratosMultiphysics/Kratos/blob/c40c51f379cda3501472d00c90fc7413ac543db2/kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h#L854-L856